src/odemis/driver/keysight.py

```diff
@@ -0,0 +1,333 @@
+            try:
+                self._accesser = IPBusAccesser(address)
+            except Exception as e:
+                logging.info("Could not establish connection to the device through IP bus accesser: %s", e)
+                raise
+
+            return address
+
+        if address == "/dev/fake":
+            return address
+
+        else:
+            raise HwError("Failed to find a device on the address: '%s'. "
+                          "Check that the device is turned on and connected to "
+                          "the computer." % (address,))
+
+    def _sendCommand(self, cmd):
+        """
+        cmd (str): command to be sent to device
+        """
+        # Send the given command to the instrument
+        logging.debug("Sending command %s", to_str_escape(cmd))
+        self._accesser.sendCmd(cmd)  # TODO: check with real hw if a brief wait needed(like 50 ms of time.sleep)
+
+    def _sendQuery(self, q):
+        """
+        q (str): query to be sent to device
+        """
+        # Send the given command to the instrument
+        logging.debug("Sending query %s", to_str_escape(q))
+        self._accesser.sendQuery(q)  # TODO: check with real hw if a brief wait needed(like 50 ms of time.sleep)
+
+    def setPeriod(self, c, p):
+        """
+        c (int): the channel to set the period of. 1 or 2
+        p (str): the period value
+        """
+        # default will be 4000000 Hz (40 MHz) = 4e-6 for period val
+        # it will be fixed to 40mhz for sync scan and will be adjustable for single sweep or labcube
+
+        f = float(1 / p)  # the instrument accepts frequency as input so period is inverted
+        self._sendCommand("sour%d:freq %f" %(c, f))
+
+    def setVoltageMin(self, c, v):
+        """
+        c (int): the channel to set the low voltage. 1 or 2
+        v (int): the voltage value
+        """
+        self._sendCommand("sour%d:volt:low %d" % (c, v))
+
+    def setVoltageMax(self, c, v):
+        """
+        c (int): the channel to set the period of. 1 or 2
+        p (str): the period value
+        """
+        self._sendCommand("sour%d:volt:high %d" % (c, v))
+
+    def setVoltageLimitMin(self, c, v):
+        """
+        c (int): the channel to set the period of. 1 or 2
+        p (str): the period value
+        """
+        self._sendCommand("sour%d:volt:lim:low %d" % (c, v))
+
+    def setVoltageLimitMax(self, c, v):
+        """
+        c (int): the channel to set the period of. 1 or 2
+        p (str): the period value
+        """
+        self._sendCommand("sour%d:volt:lim:high %d" % (c, v))
+
+    def setDutyCycle(self, c, dc):
+        """
+        c (int): the channel to set the period of. 1 or 2
+        p (str): the period value
+        """
+        self._sendCommand("sour%d:func:squ:dcyc %s" % (c, dc))
+
+    def setTracking(self, c, t):
+        """
+        c (int): the channel to set the period of. 1 or 2
+        p (str): the period value
```

Comment:
Wrong comment. -> bool?

Reply:
the tracking mode can be ON, OFF or INV so should still be string but the `p` param in the docstring needs an update obviously.